### PR TITLE
fix bug on isUnderFileRoot

### PR DESCRIPTION
**Pull request by JackTn**

### CODE REVIEW

File: lib/swagger/fileLoader.ts [hunk 1]
```diff
@@ -79,11 +79,14 @@ export class FileLoader implements Loader<string> {
   }
 
   public isUnderFileRoot(filePath: string) {
-    if (!this.opts.fileRoot) {
+    if (this.opts.fileRoot === undefined || this.opts.fileRoot === null) {
+      return false;
+    } else if (!this.opts.fileRoot) {
```
Comment (raych1) [inline lib/swagger/fileLoader.ts]: Line#84 and line #85 aren't needed any more.

**Comment by raych1** [comment]
@mikekistler , can you get this PR branch and validate if it's fixed in macos? Shanghai team cannot find a mac :-(

**Comment by mikekistler** [comment]
This branch does not fix the problem. I created a test that should help you verify the fix on non-Mac systems:

https://github.com/mikekistler/oav/blob/file-loader-test/test/swagger/fileLoaderTests.ts

**Comment by mikekistler** [comment]
What's the status on this fix?

**Comment by JackTn** [comment]
@mikekistler  can you try this test for this branch again ? Thanks !

File: test/swagger/fileLoaderTest.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import assert from "assert";
+import { FileLoader, FileLoaderOption } from "../../lib/swagger/fileLoader";
+
+describe("FileLoader functions", () => {
+  describe("isFileUnderRoot", () => {
```
Comment (mikekistler) [inline test/swagger/fileLoaderTest.ts]:
```suggestion
  describe("isUnderFileRoot", () => {
```

File: lib/swagger/fileLoader.ts [hunk 3]
```diff
@@ -79,11 +79,15 @@ export class FileLoader implements Loader<string> {
   }
 
   public isUnderFileRoot(filePath: string) {
-    if (this.opts.fileRoot === undefined || this.opts.fileRoot === null) {
-      return false;
+    if (!this.opts.fileRoot) {
```
Comment (raych1) [inline lib/swagger/fileLoader.ts:82]: I think line#82-#84 is not needed. No matter opts.fileRoot is provided or not, we need to check if the path is under root.

Comment (JackTn) [reply lib/swagger/fileLoader.ts:82]: if this.opts.fileRoot's type is string | undefined. If fileRoot is undefined, the path is under root or not?

**Review by mikekistler — APPROVED** [review]
These changes work on Mac. Thanks.

File: lib/swagger/fileLoader.ts (hunk 3 above)
Comment (JackTn) [reply lib/swagger/fileLoader.ts:82]: one question. In origin code, if opts.fileRoot is undefined, it return true. But now, it always retrun false when opts.fileRoot is undefined. Is it correct ? @raych1

Comment (raych1) [reply lib/swagger/fileLoader.ts:82]: Let's keep original behavior.

Comment (JackTn) [reply lib/swagger/fileLoader.ts:82]: ok and it can be merged now @raych1

**Review by raych1 — APPROVED** [review]

**Pull request merged**